src/schism/schism_esmf_util.F90

```diff
@@ -573,8 +579,17 @@ subroutine SCHISM_StateUpdate1(state, name, farray, kwe, isPtr, rc)
 
   if (intent == ESMF_STATEINTENT_IMPORT) then 
 
-    do ip = 1, isPtr%numOwnedNodes
-      farray(isPtr%ownedNodeIds(ip)) = farrayPtr1(ip)
+    ! all nodes that construct the element will get same value
```

Comment:
Before accepting this change, we have to consider why we would project node vars onto elems. Shouldn't we rather just make sure that the size of farray is matching with the node / elem based vars and then based on meshloc info **not** project?

Reply:
@platipodium I placed if statements to the function to work differently in node vs element. I did not check the node version yet but I'll do. Please wait for my confirmation that node (with ESMf connector) is also working as expected before merging the PR.